src/OrchardCore.Modules/OrchardCore.Queries/Sql/Deployment/QueryBasedContentDeploymentSource.cs

```diff
@@ -0,0 +1,95 @@
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using OrchardCore.ContentManagement;
+using OrchardCore.Deployment;
+
+namespace OrchardCore.Queries.Sql.Deployment
+{
+    public class QueryBasedContentDeploymentSource : IDeploymentSource
+    {
+        private readonly IQueryManager _queryManager;
+
+        public QueryBasedContentDeploymentSource(IQueryManager queryManager)
+        {
+            _queryManager = queryManager;
+        }
+
+        public async Task ProcessDeploymentStepAsync(DeploymentStep step, DeploymentPlanResult result)
+        {
+            var contentStep = step as QueryBasedContentDeploymentStep;
+
+            if (contentStep == null)
+            {
+                return;
+            }
+
+            var data = new JArray();
+
+            var query = await _queryManager.GetQueryAsync(contentStep.QueryName) as SqlQuery;
```

Comment:
I mistakenly assumed this was supposed to be for SQL queries only.

Reply:
Due to the abstraction, it should be straightforward to make it compatible with any Query. Since a site may use Lucene, SQL, or any future/custom type of Queries, that would be quite useful.